src/satisfia/agents/makeMDPAgentSatisfia.py

```diff
@@ -815,54 +817,57 @@ def randomTieBreaker(self, state, action):
 	# now we can combine all of the above quantities to a combined (safety) loss function:
 
 	# state, action, aleph4state, aleph4action, estActionProbability
-	#@lru_cache(maxsize=None)
+	@lru_cache(maxsize=None)
```

Comment:
this was disabled but it seems to help.

was there a reason why you disabled it?

If it was a memory problem, it might be worth to put a limit, like 1024: it would probably still help